helios-client/src/main/java/com/spotify/helios/client/AuthenticatingHttpConnector.java

```diff
@@ -105,8 +111,11 @@ public HttpURLConnection connect(final URI uri, final String method, final byte[
     try {
       log.debug("connecting to {}", ipUri);
 
-      if (clientCertificatePath.isPresent()) {
-        // prioritize using the certificate file if set
+      if (accessTokenOpt.isPresent()) {
+        // prioritize using bearer token
+        return connectWithBearerToken(ipUri, method, entity, headers);
+      } else if (clientCertificatePath.isPresent()) {
```

Comment:
Yeah, they're not mutually exclusive at all. I'm not sure there's much value in doing both though. So long as the master is https, the client cert is only used for authentication, and examining the token seems like a better method.